api/views.py

```diff
@@ -67,6 +67,9 @@ def create(self, request, **kwargs):
                 puzzle.answer = answer.text
                 if puzzle.chat_room:
                     puzzle.chat_room.archive_channels()
+                    msg = f"{puzzle.name} has been solved with {answer.text}!"
+                    puzzle.chat_room.announce(msg)
```

Comment:
This is kinda confusing to read because it looks like you're making an announcement to the chat room, wonder if `puzzle.chat_room.service().announce(msg)` is any clearer